ci-operator/step-registry/openshift/microshift/e2e/greenboot/openshift-microshift-e2e-greenboot-commands.sh

```diff
@@ -0,0 +1,167 @@
+if [[ -z "${GOOGLE_COMPUTE_ZONE}" ]]; then
+  echo "Expected \${SHARED_DIR}/openshift_gcp_compute_zone to contain the GCP zone"
+  exit 1
+fi
+
+mkdir -p "${HOME}"/.ssh
+mock-nss.sh
+
+# gcloud compute will use this key rather than create a new one
+cp "${CLUSTER_PROFILE_DIR}"/ssh-privatekey "${HOME}"/.ssh/google_compute_engine
+chmod 0600 "${HOME}"/.ssh/google_compute_engine
+cp "${CLUSTER_PROFILE_DIR}"/ssh-publickey "${HOME}"/.ssh/google_compute_engine.pub
+echo 'ServerAliveInterval 30' | tee -a "${HOME}"/.ssh/config
+echo 'ServerAliveCountMax 1200' | tee -a "${HOME}"/.ssh/config
+chmod 0600 "${HOME}"/.ssh/config
+
+cp "${CLUSTER_PROFILE_DIR}"/pull-secret "${HOME}"/pull-secret
+
+gcloud auth activate-service-account --quiet --key-file "${CLUSTER_PROFILE_DIR}"/gce.json
+gcloud --quiet config set project "${GOOGLE_PROJECT_ID}"
+gcloud --quiet config set compute/zone "${GOOGLE_COMPUTE_ZONE}"
+gcloud --quiet config set compute/region "${GOOGLE_COMPUTE_REGION}"
+
+gcloud compute firewall-rules update "${INSTANCE_PREFIX}" --allow tcp:22,icmp,tcp:80
+
+cat << 'EOFEOF' > "${HOME}"/greenboot_check.sh
+#!/usr/bin/env bash
+
+set -euo pipefail
+IFS=$'\n\t'
+set -x
+
+function check_greenboot_exit_status() {
+  local expectedRC=\$1
+
+  echo 1 | sudo microshift-cleanup-data --all
+  sudo systemctl enable --now microshift || true
+
+  for check_script in $(find /etc/greenboot/check/ -name \*.sh | sort) ; do
+    echo Running \${check_script}...
+    local currentRC=1
+    if sudo \${check_script} ; then
+      currentRC=0
+    fi
+
+    if [ \${currentRC} != \${expectedRC} ] ; then
+      exit 1
+    fi
+  done
+}
+
+#
+# Initial check must succeed (set timeout of 60s to speed up the process)
+#
+sudo tee /etc/greenboot/greenboot.conf &>/dev/null <<EOF
+MICROSHIFT_WAIT_TIMEOUT_SEC=60
+EOF
+check_greenboot_exit_status 0
+
+#
+# User workload health
+# See https://github.com/openshift/microshift/blob/main/docs/greenboot_dev.md#user-workload-health
+#
+MANIFEST_DIR=/etc/microshift/manifests
+sudo mkdir -p \${MANIFEST_DIR}
+
+sudo tee \${MANIFEST_DIR}/busybox.yaml &>/dev/null <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: busybox
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: busybox-deployment
+spec:
+  selector:
+    matchLabels:
+      app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: busybox
+        image: BUSYBOX_IMAGE
+        command:
+          - sleep
+          - "3600"
+EOF
+
+sudo tee \${MANIFEST_DIR}/kustomization.yaml &>/dev/null <<EOF
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: busybox
+resources:
+  - busybox.yaml
+images:
+  - name: BUSYBOX_IMAGE
+    newName: registry.k8s.io/busybox
+EOF
+
+SCRIPT_FILE=/etc/greenboot/check/required.d/50_busybox_running_check.sh
+sudo curl -s https://raw.githubusercontent.com/openshift/microshift/main/docs/config/busybox_running_check.sh -o \${SCRIPT_FILE}
```

Comment:
This is dangerous. If this job needs to run from a release branch it is going to take the script from main.
You should be able to copy this file from the src container into the one this ref is using (microshift-validate).